src/routines/mpilib20_reduce.F90

```diff
@@ -0,0 +1,83 @@
+!> Wrapping of the MPI_REDUCE methods
+!> TODOs
+!> Unravel how MPI_Datatype operates
+!> Overload for MPI_IN_PLACE?
+!> Testing
+module mpilib20_reduce_m
+
+    use mpilib20_init_finalise, only : mpi_env_type
+
+    implicit none
+
+    private
+
+    interface mpilib20_reduce
+        module procedure mpilib20_reduce_int_scalar
+        module procedure mpilib20_reduce_int_vec
+    end interface
+    
+    public :: mpilib20_REDUCE
+    
+contains
+
+    !> MPI_REDUCE WRAPPER: INTEGER
+    subroutine mpilib20_reduce_int_scalar(sendbuf, recvbuf, operation, mpi_env, root)
+        use mpi_bindings, only : MPI_REDUCE, MPI_Op, &
+            MPI_Datatype, MPI_INTEGER
+        use mpilib20_init_finalise, only : root_id
+        !> Variable containing set to be sent
+        type(MPI_Datatype),          intent(in)      :: sendbuf
```

Comment:
Send and receive buffers should be integers in this case - they're not tided to MPI types.